### PR TITLE
Kvc lb151.ag mirca ctry c glu irr

### PR DESCRIPTION
**Pull request by kvcalvin**
@bpbond: This is the branch that is failing the old/new test on chunks unrelated to mine. The four chunks that fail are: `L100.FAO_fallowland_kha`, `L100.FAO_harv_CL_kha`, `L100.gdp_mil90usd_ctry_Yh`, and `L101.Pop_thous_GCAM3_ctry_Y`. I didn't have a problem on my old computer, but new computer doesn't like those.

### CODE REVIEW

File: R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R [hunk 1]
```diff
@@ -43,73 +46,138 @@ module_aglu_LB151.ag_MIRCA_ctry_C_GLU_irr_DISABLED <- function(command, ...) {
...
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Compute shares of irrigated and rainfed harvested area from MIRCA data
+    # Shares are by country (iso), land unit (GLU), and MIRCA_crop (different from GTAP crop or GCAM_commodity)
+    # NOTE: MIRCA data is for a single year. We use the shares, but not the actual land areas.
+     L100.MIRCA_irrHA_ha %>%
+      mutate( MIRCA_crop = paste0( "Crop", MIRCA_crop )) %>%                    # Convert name of crop to include "Crop"
```
Comment (bpbond) [inline R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R]: If not too much of a pain please remove GCAM-style spacing by parentheses and brackets.

File: R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R [hunk 2]
```diff
@@ -43,73 +46,138 @@ module_aglu_LB151.ag_MIRCA_ctry_C_GLU_irr_DISABLED <- function(command, ...) {
...
+      select(-Yield_kgHa_irrigated, -Yield_kgHa_rainfed) %>%
+      mutate(yieldratio=if_else( is.na(yieldratio), 1, yieldratio )) %>%                                                # Replace NAs with 1
+      mutate(yieldratio=if_else( yieldratio==0, 1, yieldratio )) %>%                                                    # Replace zeros with 1
+      mutate(yieldratio=if_else( is.infinite(yieldratio), 1, yieldratio ))  ->                                          # Replace Infs with 1
+      Yieldratio_ctry_crop
+
+    # Now, use the yield ratio to solve for the production shares (irrigated versus rainfed)
+    # This calculation preserves the harvested area shares and yield ratios computed above.
+    ag_HA_ha_ctry_crop %>%
+      left_join( Yieldratio_ctry_crop[ c( "iso", "GLU", "GTAP_crop", "yieldratio") ], by = c( "iso", "GLU", "GTAP_crop") ) %>%  # Map in the yield ratio
```
Comment (bpbond) [inline R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R]: For consistent style might do `left_join(select(Yieldratio_ctry_crop, iso, ...), by = ...)`     ...not critical though

File: R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R [hunk 3]
```diff
@@ -43,73 +46,138 @@ module_aglu_LB151.ag_MIRCA_ctry_C_GLU_irr_DISABLED <- function(command, ...) {
...
+      mutate(irrshareHA=irrHA/(irrHA + rfdHA)) %>%                                                                              # Compute the share of irrigated harvested area
+      mutate(irrshareProd=(irrshareHA*yieldratio)/((irrshareHA*yieldratio)+(1-irrshareHA))) %>%                                 # Compute the share of irrigated production
```
Comment (bpbond) [inline R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R]: Add spaces around mathematical operators pls

File: R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R [hunk 4]
```diff
@@ -43,73 +46,138 @@ module_aglu_LB151.ag_MIRCA_ctry_C_GLU_irr_DISABLED <- function(command, ...) {
...
+    L151.ag_rfdHA_ha_ctry_crop %>%
+      add_title("Irrigated harvested area by country, GTAP crop, GLU") %>%
+      add_units("Hectares") %>%
+      add_comments("Uses shares of rainfed area from MIRCA, combined with total harvested area, to calculate rainfed harvested area") %>%
+      add_comments("Data is for a single year (circa 2000)") %>%
       add_legacy_name("L151.ag_rfdHA_ha_ctry_crop") %>%
-      add_precursors("precursor1", "precursor2", "etc") %>%
-      # typical flags, but there are others--see `constants.R`
-      add_flags(FLAG_NO_TEST, FLAG_LONG_YEAR_FORM, FLAG_NO_XYEAR) ->
+      add_precursors("aglu/FAO_ag_items_PRODSTAT","L100.LDS_ag_HA_ha","L100.MIRCA_irrHA_ha","L100.MIRCA_rfdHA_ha") ->
```
Comment (bpbond) [inline R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R]: Two questions: do all the outputs use ALL the inputs? Please check and modify if necessary. Also, remember about `same_precursors_as()` which can save typing and make things clearer.

Comment (kvcalvin) [inline R/zchunk_LB151.ag_MIRCA_ctry_C_GLU_irr.R]: Thanks for that. I realized when going through this that one of the inputs to this file is never used. Should I just remove that?

**Comment by swaldhoff** [comment]
Thanks for the clarification, Kate. That makes much more sense! 
It would still be worthwhile to remove the (pure) duplicates from the AGLU_ctry file and add some documentation for the CROSIT data, but we can open issues on that if it's easier.

**Comment by swaldhoff** [comment]
Clarification, I meant removing the duplicates in the csv file so you don't need to correct it in the code here.

**Pull request merged**